### PR TITLE
Don't programmatically lowercase post type label

### LINKED ISSUE #49587: "Link to […]" should not lowercase post type label
### Description

In #40378, the string "Link to %s" (e.g. "Link to page") of the "Post Featured Image" block was made translatable. To ensure proper English grammar and because the WordPress API does not include support for lowercase post type labels, the post type's singular name is passed to `toLowerCase()`.

While this works for English, it causes incorrect Grammar for languages that capitalize nouns, e.g. German. There, this string becomes "Link zu seite" which is wrong. It should be "Link zu Seite".

Generally, capitalization should never be changed programmatically. It may work for English, but certainly not German and probably a couple of other languages.

### Step-by-step reproduction instructions

1. Switch the locale to German
2. Insert the "Post Featured Image" to a page (not post)
3. Open the block settings
4. Verify that the setting incorrectly reads "Link zu seite"

### Screenshots, screen recording, code snippet

Bildschirmfoto 2023-04-04 um 21 08 32

### Environment info

WordPress 6.2
no Gutenberg plugin

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

### PR DESCRIPTION
**Pull request by tyrann0us**
## What?
Remove the programmatically lowercasing of the post type label in the "Link to %s" setting of the "Post Featured Image" block.

Fixes #49587 

## Why?
Programmatically lowercasing the label causes wrong grammar in languages that capitalize nouns like German.

## How?
The `.toLowerCase()` call is removed. While this leads to capitalized post type labels also in English, IMO this acceptable. On multiple occassions throughout WordPress, English nouns are already capitalized, e.g., the "Update Profile" button on the user profile page:
Bildschirmfoto 2023-04-04 um 21 43 40

Also, the translators comment is changed to use "Page" because a) "post" is already used in the fallback string and b) the label is now capitalized.

## Testing Instructions
1. Insert the "Post Featured Image" to a page (not post)
2. Open the block settings
3. Verify that the setting  reads "Link to Page" instead of "Link to page"
4. Switch the locale to German
5. Verify that the setting reads "Link zu Seite" instead of "Link zu seite"

### Testing Instructions for Keyboard
Does not apply.

## Screenshots or screencast 
Before:
Bildschirmfoto 2023-04-04 um 21 08 32

After:
Bildschirmfoto 2023-04-04 um 21 45 39

### CODE REVIEW

**Review by aaronrobertshaw — COMMENTED** [review]
Thanks for putting together the PR to address this @tyrann0us 👍 

I understand certain languages have strict rules regarding capitalization. 

I'm not 100% sure simply stripping the `toLowerCase()` call here is the best option. There was a deliberate decision around control text being sentence cased (for English). @WordPress/gutenberg-design might be able to provide clarity here both in terms of direction and scope for whether removing the lowercase transform is fine.

The lowercase post type name is being passed through for translation which would allow translators to adjust the resulting text's capitalization e.g. from "page" to "Seite".

My only reservation on relying solely on translations would be that it could be possible for a custom post type to be registered with a non-English name and may not then have a translation to reinstate any capitalization.

My apologies for holding up this PR but it would be good to see if we can satisfy the needs of multiple languages rather than compromise one for the other.

**Comment by jasmussen** [comment]
> I'm not 100% sure simply stripping the toLowerCase() call here is the best option. There was a deliberate decision around control text being sentence cased (for English). 

Thanks for the ping. Yes as a rule of thumb when authoring new contents in the default English language, we don't use titlecase for form controls. That's mainly title case, though, and as shown in the example, proper nouns in German are capitalized, so it seems incorrect to programmatically lowercase it. So I wouldn't personally mind removing that function for this particular use case.

The lowercase-only, in my best understanding, is a guideline for full phrases (such as `__( 'Link to homepage' )`), and in cases where those are pieced together we shouldn't transform them (i.e. `__( 'Link to %s' )`).

That being said, concatenating labels like these in general seems to be something to avoid. In the case of the link toggle, it is understandably necessary since featured images can sit on posts or pages etc. But whenever it's possible, it seems best for translation purposes to avoid concatenation, so the full context can be available to the translator.

**Comment by tyrann0us** [comment]
Thank you for the feedback!

@aaronrobertshaw
> The lowercase post type name is being passed through for translation which would allow translators to adjust the resulting text's capitalization e.g. from "page" to "Seite".

I might be overlooking something, but since the post type label is _not_ passed to the translation function, but only referenced as a placeholder, how would a translator be able to change its capitalization? 🤔

**Review by aaronrobertshaw — APPROVED** [review]
> I might be overlooking something, but since the post type label is not passed to the translation function, but only referenced as a placeholder, how would a translator be able to change its capitalization? 🤔

I misread that code sorry, you are correct 🤦 

> So I wouldn't personally mind removing that function for this particular use case.

It looks like we have a green light to remove the lowercase call.

I've taken this for a spin and it tests as advertised.

LGTM 🚢

**Comment by aaronrobertshaw** [comment]
The failing e2e appears unrelated, so I've restarted that job.

Assuming it passes, I'll merge this before finishing up for the day.

**Review by aristath — APPROVED** [review]

**Pull request merged**
